sbnci/Modules/RecoEff/RecoEff_module.cc

```diff
@@ -0,0 +1,417 @@
+
+    for(auto particle : particles){
+      if(particle->Mother() != 0 || particle->StatusCode() != 1) continue;
+
+      mc_trackID = -999; mc_PDG = -999; mc_x0 = -999; mc_y0 = -999; mc_z0 = -999;
+      mc_xEnd = -999; mc_yEnd = -999; mc_zEnd = -999; mc_pX0 = -999; mc_pY0 = -999;
+      mc_pZ0 = -999; mc_energy0 = -999; mc_momentum = -999; mc_pXEnd = -999; mc_pYEnd = -999;
+      mc_pZEnd = -999; mc_energyEnd = -999; mc_mass = -999; mc_theta_xy = -999; mc_theta_yz = -999;
+      mc_theta_xz = -999; mc_length = -999; reco_nTracks = -999; reco_nShowers = -999; 
+      reco_shower_purity = -999; reco_shower_completeness = -999; reco_track_purity = -999; 
+      reco_track_completeness = -999; reco_track_length = -999; reco_shower_dEdx = -999; 
+      reco_isReconstructed = false;
+
+      mc_trackID = particle->TrackId();
+      mc_PDG = particle->PdgCode();
+      mc_x0 = particle->Vx();
+      mc_y0 = particle->Vy();
+      mc_z0 = particle->Vz();
+
+      float xOutEdge = 185, xInEdge = 1.5, yEdge = 185, zFrontEdge = 30, zBackEdge = 435;
+      if(mc_PDG == 11 || mc_PDG == -11) {
+	xOutEdge = 175, xInEdge = 1.5, yEdge = 175, zFrontEdge = 30, zBackEdge = 450;}
+
+      if(TMath::Abs(mc_x0) > xOutEdge || TMath::Abs(mc_x0) < xInEdge ||
+	 TMath::Abs(mc_y0) > yEdge || mc_z0 < zFrontEdge || mc_z0 > zBackEdge) continue;
+
+      mc_xEnd = particle->EndX();
+      mc_yEnd = particle->EndY();
+      mc_zEnd = particle->EndZ();
+      mc_pX0 = particle->Px();
+      mc_pY0 = particle->Py();
+      mc_pZ0 = particle->Pz();
+      mc_energy0 = particle->E();
+      mc_momentum = particle->P();
+      mc_pXEnd = particle->EndPx();
+      mc_pYEnd = particle->EndPy();
+      mc_pZEnd = particle->EndPz();
+      mc_energyEnd = particle->EndE();
+      mc_mass = particle->Mass();
+      mc_theta_xy = TMath::RadToDeg() * TMath::ATan(mc_pX0/mc_pY0);
+      mc_theta_yz = TMath::RadToDeg() * TMath::ATan(mc_pY0/mc_pZ0);
+      mc_theta_xz = TMath::RadToDeg() * TMath::ATan(mc_pX0/mc_pZ0);
+      mc_length = TrueTrackLength(particle);
+
+      if(n_tracks_map[mc_trackID] > 0 || n_showers_map[mc_trackID] > 0) reco_isReconstructed = true;
+      reco_nTracks = n_tracks_map[mc_trackID];
+      reco_nShowers = n_showers_map[mc_trackID];
+      reco_shower_purity = shower_pur_map[mc_trackID];
+      reco_shower_completeness = shower_comp_map[mc_trackID];
+      reco_track_purity = track_pur_map[mc_trackID];
+      reco_track_completeness = track_comp_map[mc_trackID];
+      reco_track_length = track_length_map[mc_trackID];
+      reco_shower_dEdx = shower_dEdx_map[mc_trackID];
+
+      fParticleTree->Fill();
+    }
+  }
+}
+
+std::vector<art::Ptr<recob::PFParticle> > RecoEff::GetPrimaryPFPs()
+{
+  std::vector<art::Ptr<recob::PFParticle> > primaries;
+  for(unsigned int pfp_i = 0; pfp_i < eHandlePFPs->size(); ++pfp_i) {
+    const art::Ptr<recob::PFParticle> pfp(eHandlePFPs,pfp_i);
+    if(pfp->IsPrimary()){
+      if(pfp->PdgCode() == 12 || pfp->PdgCode() == -12
```

Comment:
if (std::abs(pdg) == 12) would make this more succinct.

Reply:
Done [here](https://github.com/SBNSoftware/sbnci/pull/7/commits/e39fb816cd595c1be5af6af8f168c73d6c6e0fc1)